admin/onpage/class-onpage-email-presenter.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * @package WPSEO\Admin
+ */
+
+/**
+ * This class is the presenter for the email. Based on the given parameters and data it will parse the email message
+ * that will be send
+ */
+class WPSEO_OnPage_Email_Presenter {
+
+	/**
+	 * @var string The generated messsage.
+	 */
+	private $message;
+
+	/**
+	 * @var array The old and the new status as an array.
+	 */
+	private $statuses;
+
+	/**
+	 * @var array The translated values for the indexability statuses.
+	 */
+	private $status_values;
+
+	/**
+	 * Constructing the object
+	 *
+	 * @param array $statuses Array with the statusses, normally it are the old and the new status.
+	 */
+	public function __construct( array $statuses ) {
+		$this->status_values = $this->get_status_values();
+		$this->statuses      = $statuses;
+		$this->message       = $this->generate_message();
+	}
+
+	/**
+	 * Returns the generated message.
+	 *
+	 * @return string
+	 */
+	public function get_message() {
+		return $this->message;
+	}
+
+	/**
+	 * Mapping the statusses to translated values
+	 */
+	private function get_status_values() {
+		return array(
+			'0' => __( 'not indexable', 'wordpress-seo' ),
+			'1' => __( 'indexable', 'wordpress-seo' ),
```

Comment:
I wonder if it's better to have full messages instead of statuses.

`The indexability from your website %1$s, went from %2$s to %3$s` is harder to translate then `The indexability of your website %1$s, went from not indexable to indexable.` This could also be an opportunity to use an even better message such as `Good job! Your site %1$s is now indexable when it previously wasn't.` or some other variation.